### PR TITLE
Fix/input extraction

### PR DESCRIPTION
**Pull request by brysonjbest**
Fix handling of 'today' for validations and generating of scenarios.
Add additional cleaning to validation inputs to allow for improved matching.
Increase variability of items generated using array of text options from validations.
Fix capturing of unique inputs for accurately displaying used/unused inputs on the BRM App.

### CODE REVIEW

File: src/api/scenarioData/scenarioData.service.ts [hunk 1]
```diff
@@ -288,9 +301,9 @@ export class ScenarioDataService {
         return scenarios;
 
       case 'number-input':
-        const numberValues = validationCriteria?.split(',').map((val: string) => val.trim());
+        const numberValues = validationCriteria?.split(',').map((val: string) => this.cleanValue(val));
         const minValue = (numberValues && parseInt(numberValues[0], 10)) || 0;
-
+        console.log(numberValues, 'number values');
```
Comment (timwekkenbc) [inline src/api/scenarioData/scenarioData.service.ts:306]: Console log left in (two more further below as well)

File: src/api/decisions/validations/validations.service.ts [hunk 2]
```diff
@@ -269,6 +277,15 @@ export class ValidationService {
     }
   }
 
+  private normalizeText(text: string): string {
+    return text
+      ?.trim()
+      ?.replace(/[\[\]()]/g, '')
+      ?.replace(/[''"]/g, '')
+      ?.replace(/\s+/g, ' ')
+      ?.trim();
```
Comment (timwekkenbc) [inline src/api/decisions/validations/validations.service.ts:286]: I suck at regex, but I think you can shorten it to:
```
text
  ?.trim()
  ?.replace(/[\[\]()'"‘’“”]/g, '')
  ?.replace(/\s+/g, ' ');
```
Not sure you need any of the double `trim` functions on any of the `cleanValue` functions either.

**Review by timwekkenbc — APPROVED** [review]
Looks good

File: src/api/scenarioData/scenarioData.service.ts (hunk 1 above)
Comment (brysonjbest) [reply src/api/scenarioData/scenarioData.service.ts:306]: Thank you! Forgot cleanup.

**Pull request merged**